### PR TITLE
Add switch example using HTML checkbox input

### PR DESCRIPTION
**Pull request by jongund**
[Preview link](https://raw.githack.com/w3c/aria-practices/switch-checkbox/examples/switch/switch-checkbox.html)

* [x] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by @mcking65
* [x] [Functional review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#functional_review) by @carmacleod 
* [x] [Visual design review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#design_review) by @a11ydoer 
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review) on Windows/Mac OS by APG members on Aug 3.
* [x] [Code review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#code_review) by @jesdaigle / @zcorpan
* [x] [Test review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) by @jesdaigle / @zcorpan

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/1895.html" title="Last updated on Nov 10, 2021, 4:15 PM UTC (558fc72)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/1895/b0c809b...558fc72.html" title="Last updated on Nov 10, 2021, 4:15 PM UTC (558fc72)">Diff</a>

### CODE REVIEW

**Comment by jongund** [comment]
@mcking65 @jnurthen 
If you are using an `input[type=checkbox]` with the `switch` role, do you need to use `aria-checked`?
The W3C validator is requiring `aria-checked`.

**Comment by jnurthen** [comment]
No - see the note at https://w3c.github.io/aria/#requiredState

**Comment by jongund** [comment]
@mcking65 
For some reason this branch is failing the menu-button link test for the ENTER key, even though none of the menu button example files were touched in this pull request.

**Comment by shirsha** [comment]
I would suggest not to use CSS generated content as it is not supported screen readers on older versions, it is difficult to localize, difficult to change style by the user.

**Comment by carmacleod** [comment]
Functional review:

Chrome, Firefox, and Safari:
- If I double-click or triple-click the switch, then the label text gets selected, which looks a bit funny. I think maybe the label needs to have CSS `user-select: none;` so that this won't happen. https://developer.mozilla.org/en-US/docs/Web/CSS/user-select

Safari:
- If I click and hold the switch, the label text turns white. (Safari only)

**Comment by shirsha** [comment]
If you use user-select: none; then VO/safari ignores the control. 
https://bugs.webkit.org/show_bug.cgi?id=82692

**Comment by jongund** [comment]
@mcking65 
Made updates to:
1. Move on/off labels from CSS to content, may need `aria-hidden`.
2. Added no selection, so text is selected when activating with a mouse.

File: examples/switch/switch-checkbox.html [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    This example implements the <a href="../../#switch">Switch design pattern</a> using <code>input[type=&quot;checkbox&quot;]</code> elements and CSS borders to implement the graphical rendering of the state of the switch.  Switches are similar to checkboxes, but should be rendered by assistive technologies as "on" and "off", instead of "checked" and "unchecked".
+  </p>
+  <p>Similar examples include: </p>
+  <ul>
+    <li><a href="switch.html">Switch example using the <code>div</code> element.</a></li>
+    <li><a href="switch-button.html">Switch example using the <code>button</code> element.</a>.</li>
+  </ul>
+
+  <section>
+    <h2 id="ex_label">Example</h2>
```
Comment (zcorpan) [inline examples/switch/switch-checkbox.html:39]: Missing `<div class="example-header">` etc to enable the CodePen button.
https://github.com/w3c/aria-practices/wiki/How-to-add-the-Open-In-Codepen-button-to-an-example-page#steps-to-add-the-button-to-existing-examples

**Review by zcorpan — CHANGES_REQUESTED** [review]
Code review: 1 comment, otherwise LGTM
Test review: LGTM

File: examples/switch/switch-checkbox.html (hunk 1 above)
Comment (jongund) [reply examples/switch/switch-checkbox.html]: @zcorpan 
The code pen button is working now in the preview view

**Comment by jongund** [comment]
@mcking65 
Removed the keyboard event handler to support the enter key.

**Review by a11ydoer — APPROVED** [review]
Thanks for reflecting the consensus from APG meeting today.

**Comment by mcking65** [comment]
@jesdaigle could you please re-review; Simon requested changes. I believe Jon made the requested changes.

**Comment by mcking65** [comment]
@jongund 

I've completed editorial review and made a few changes.

One part where I feel we don't quite have enough explanation of the "why" is the documentation about borders and spacing for HCM. We have:

>Spacing and borders are used to ensure the graphics will be visible and discernible when browser or operating system high contrast settings are enabled:
>   * Two pixel borders are used for the switch state container and the circle indicating the on and off position of the switch. 
>   * There are 2 pixels of space between the container border and the border of the circle. 

How do the spacing and borders impact HCM? What if the spacing were different? How might someone make something that looks similar that wouldn't work in HCM? This documentation says why we want spacing and borders but does not explain why this spacing and these borders work and why other implementation choices would not work. What is it that makes these specific implementation choices work in HCM and other possible choices not work?

**Comment by jongund** [comment]
@mcking65 
I have updated and added additional information to the accessibility features.  Please review when you have a chance, if you approve of the changes I will update the other switch examples.

**Comment by jongund** [comment]
@zcorpan 
Not sure what to do, it seems there have been some changes in the commit process and it modifies the `examples
\index.html` which creates a merge block for my pull request.  Since I don't touch the `examples\index.html` file not sure how to fix this.

**Comment by a11ydoer** [comment]
@jesdaigle will help to resolve this issue.

**Comment by jongund** [comment]
@mcking65 
I took out the statement about using `input.checked` property as a synchronizer for the graphical state.  I think that was what you had concerns about.

**Review by mcking65 — APPROVED** [review]
Editorial review is complete.

**Comment by jongund** [comment]
@mcking65 
The accessibility features documentation with switch using the input element look good to me.

**Comment by mcking65** [comment]
@zcorpan, @howard-e, this has changed a fair amount since last review of code and test. Could one of you please have another look?

CC @richnoah

**Comment by zcorpan** [comment]
[`Run regression tests` CI check failed](https://github.com/w3c/aria-practices/runs/4064645727?check_suite_focus=true#step:6:190) for `menubar_menubar-navigation`, though that test was not changed in this PR. Maybe the test is flaky. I'll try rerunning.

<details>
<summary>Log output</summary>

```
menubar_menubar-navigation › after.always hook
not ok 84 - StaleElementReferenceError: The element reference of <a href="#academics"> is stale; either the element is no longer attached to the DOM, it is not in the current frame context, or the document has been refreshed
  ---
    remoteStacktrace: >
      WebDriverError@chrome://remote/content/shared/webdriver/Errors.jsm:181:5

      StaleElementReferenceError@chrome://remote/content/shared/webdriver/Errors.jsm:442:5

      element.resolveElement@chrome://remote/content/marionette/element.js:686:11

      evaluate.fromJSON@chrome://remote/content/marionette/evaluate.js:253:26

      evaluate.fromJSON@chrome://remote/content/marionette/evaluate.js:261:29

      receiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.jsm:79:29
    name: StaleElementReferenceError
    message: >-
      The element reference of <a href="#academics"> is stale; either the element is
      no longer attached to the DOM, it is not in the current frame context, or the
      document has been refreshed
    at: >-
      Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:517:15)

      parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:671:13)

      Executor.execute (node_modules/selenium-webdriver/lib/http.js:597:28)

      thenableWebDriverProxy.execute
      (node_modules/selenium-webdriver/lib/webdriver.js:729:17)
  ...
```

</details>

File: examples/switch/css/switch-checkbox.css [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+label input[role="switch"] {
+  opacity: 0;
+}
+
+label input[role="switch"] ~ .state {
+  display: inline-block;
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
+  user-select: none;
```
Comment (zcorpan) [inline examples/switch/css/switch-checkbox.css:36]: This doesn't follow https://github.com/w3c/aria-practices/wiki/Code-Guide#prefixed-properties

Should we remove that section of the code guide?

**Review by zcorpan — CHANGES_REQUESTED** [review]

**Comment by zcorpan** [comment]
Test review: LGTM

File: examples/switch/css/switch-checkbox.css (hunk 2 above)
Comment (mcking65) [reply examples/switch/css/switch-checkbox.css:36]: @zcorpan  based on our Nov 2 discussion,I think we still need this for webkit. I think @jongund is looking into whether the others can be removed from this example without issue. Do we need a separate issue for conversation about what to do for prefix guidance in the code guide?

**Comment by mcking65** [comment]
@jongund, If I recall correctly, you were going to look into removeing browser prefixes from the css if possible? I think we found out it is necessary for webkit but may not the others? Not sure if I am remembering this correctly.

**Comment by jongund** [comment]
@mcking65 Updated CSS to remove -moz-user-select and -ms-user-select properties.

**Comment by mcking65** [comment]
@zcorpan 
I think we have resolved all your feedback.
If you agree, pls change your to approve status.

File: examples/switch/css/switch-checkbox.css (hunk 2 above)
Comment (zcorpan) [reply examples/switch/css/switch-checkbox.css:34]: We can track it in https://github.com/w3c/aria-practices/issues/2110

File: examples/switch/css/switch-checkbox.css [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+label input[role="switch"] {
+  opacity: 0;
+}
+
+label input[role="switch"] ~ .state {
+  display: inline-block;
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
```
Comment (zcorpan) [inline examples/switch/css/switch-checkbox.css:33]:
```suggestion
```

**Review by zcorpan — APPROVED** [review]

**Pull request merged**